rpcserver.go

```diff
@@ -5397,3 +5404,149 @@ func (r *rpcServer) BakeMacaroon(ctx context.Context,
 
 	return resp, nil
 }
+
+// ChannelRevenue returns the revenue for a channel over a period of time.
+// To limit the size of query required to calculate revenue, channel open
+// an close times are looked up by block height, and the user provided period
+// is bounded by these values; no revenue generating events can occur before
+// channel open or after channel close.
+func (r *rpcServer) ChannelRevenue(ctx context.Context,
+	req *lnrpc.ChannelRevenueRequest) (*lnrpc.ChannelRevenueResponse, error) {
```

Comment:
Alternatively, we could expand the scope of this method into more of a "channel analysis" endpoint, so this would as is, also encompass uptime information (which can be used to compute the max effective utilization of a channel). Just a drive by thought.